driver-async/src/test/functional/com/mongodb/async/client/Fixture.java

```diff
@@ -152,6 +156,29 @@ public static void drop(final MongoNamespace namespace) {
         }
     }
 
+    public static synchronized void waitForLastServerSessionPoolRelease() {
+        if (mongoClient != null) {
+            long startTime = System.currentTimeMillis();
+            int sessionInUseCount = getSessionInUseCount();
+            while (sessionInUseCount > 0) {
+                try {
+                    if (System.currentTimeMillis() > startTime + ClusterFixture.TIMEOUT * 1000) {
+                        throw new MongoTimeoutException("Timed out waiting for server session pool in use count to drop to 0.  Now at: "
+                                + sessionInUseCount);
+                    }
+                    sleep(10);
+                    sessionInUseCount = getSessionInUseCount();
+                } catch (InterruptedException e) {
+                    throw new MongoInterruptedException("Interrupted", e);
+                }
+            }
+        }
+    }
+
+    private static int getSessionInUseCount() {
+        return mongoClient.getServerSessionPool().getInUseCount();
```

Comment:
Minor nit - put the mongoClient null check here.

Reply:
I didn't because the method is private and only called by a method that already checks for null.  (Initially this _was_ the public method, but it turned out it was racy hence the need for the public method above that waits a bit for things to settle down)